File: app/src/main/java/com/booboot/vndbandroid/bean/vndbandroid/Genre.java
package com.booboot.vndbandroid.bean.vndbandroid;

import java.util.Arrays;
import java.util.List;

/**
 * Created by od on 15/03/2016.
 */
public class Genre {
    /**
     * List of existing genres, based on MAL advanced search
     */
    private final static String[] genres_array = new String[]{
            "Action", "Adventure", "Cars", "Comedy", "Dementia", "Demons", "Drama", "Ecchi",
            "Fantasy", "Game", "Harem", "Hentai", "Historical", "Horror", "Josei", "Kids", "Magic",
            "Martial Arts", "Mecha", "Military", "Music", "Mystery", "Parody", "Police", "Psychological",
            "Romance", "Samurai", "School", "Sci-Fi", "Seinen", "Shoujo", "Shoujo Ai", "Shounen",
            "Shounen Ai", "Slice of Life", "Space", "Sports", "Super Power", "Supernatural", "Thriller",
            "Vampire", "Yaoi", "Yuri"
    };

    public final static List<String> GENRES = Arrays.asList(genres_array);

    public static boolean contains(String tagName) {
        for (String genre : GENRES) {
            if (genre.toUpperCase().equals(tagName.trim().toUpperCase())) {
                return true;
            }
        }

        return false;
    }
}

File: app/src/main/java/com/booboot/vndbandroid/model/vndbandroid/Genre.kt
package com.booboot.vndbandroid.model.vndbandroid

import java.util.*

object Genre {
    /**
     * List of existing genres, based on MAL advanced search
     */
    private val genres_array = arrayOf("Action", "Adventure", "Cars", "Comedy", "Dementia", "Demons", "Drama", "Ecchi", "Fantasy", "Game", "Harem", "Hentai", "Historical", "Horror", "Josei", "Kids",
            "Magic", "Martial Arts", "Mecha", "Military", "Music", "Mystery", "Parody", "Police", "Psychological", "Romance", "Samurai", "School", "Sci-Fi", "Seinen", "Shoujo", "Shoujo Ai",
            "Shounen", "Shounen Ai", "Slice of Life", "Space", "Sports", "Super Power", "Supernatural", "Thriller", "Vampire", "Yaoi", "Yuri")

    val GENRES = Arrays.asList(*genres_array)

    operator fun contains(tagName: String): Boolean {
        return GENRES.any { genre -> genre.toUpperCase() == tagName.trim().toUpperCase() }
    }
}